Update build instructions (#3009)
uxlfoundation/oneDAL: docs/source/get-started/build-and-run-examples.rst

      Otherwise, you need to copy :file:`examples/oneapi/dpc` and :file:`examples/oneapi/data` folders
      to the directory with right permissions. These two folders must be retained
      in the same directory level relative to each other.

>       If you want to build examples from /examples/oneapi/cpp you can set CC and CXX with other compliers as well.

Reviewer: Would this actually work if oneDAL was compiled with DPC support but then the compiler is a non-DPC one? If not, guess it's important to mention such requirement.

Reviewer: Especially since compiling oneDAL with a DPC compiler doesn't involve setting `${CC}` / `${CXX}` (it's controlled by a make argument), whereas compiling the examples does.

Author: Yes, it should but only for cpp examples

>       If you want to build examples from /examples/oneapi/cpp you can set CC and CXX with other compliers as well.

Reviewer: ```suggestion
      If you want to build examples from /examples/oneapi/cpp you can set CC and CXX with other compilers as well, but note that if oneDAL is built with DPC support (the default when using ICX), compiling the examples will require a compiler with DPC support, such as ICX.
```

>       If you want to build examples from /examples/oneapi/cpp you can set CC and CXX with other compliers as well.

Reviewer: ```suggestion
      If you want to build examples from /examples/oneapi/cpp you can set CC and CXX with other compilers as well.
```

         .. code-block:: bash

            # Navigate to examples directory and build examples
            cd /examples/oneapi/dpc
            cmake -G "Unix Makefiles⁮" -DEXAMPLES_LIST=svm_two_class_thunder # This would generate makefiles for all svm examples matching passed name
            make               # This will compile and run generated svm examples
            cmake -G "Unix Makefiles⁮" -DONEDAL_LINK=static # This wouldgenerate make for static version

>             make               # This will compile and run all the examples

Reviewer: I haven't tried the windows version, but the linux one only compiles the examples without running them.

Author: It is fine, the running step is described in the next step

      .. group-tab:: Windows

         .. code-block:: bash

            # Navigate to examples directory and build examples
            cd /examples/oneapi/dpc

>             set CC=icx

Reviewer 2: May be it makes sense to add additional section about supported compilers. Like, user can build examples with export/set CC=gcc, but without sycl features? 
I guess its better to add a comment that for CC and CXX all supported compilers can be chosen

>             set CC=icx

Reviewer: Do the examples require icx? As far as I'm aware, there shouldn't be any issue in building them with msvc if oneDAL was already built with icx.

Author: It is necessary to set this variables, otherwise the build is failed

Reviewer: Does it mean that other compilers like `msvc` are not supported for building the examples?

Author: Only for building cpp examples. To build dpcpp examples we need icx (or icpx on linux)

Reviewer: Perhaps it should clarify this in the instructions. Is it possible to build only the non-dpcpp examples?

Author: added one sentence to clarify that

>             set CXX=icx

Reviewer: ```suggestion
            set CXX=icpx
```

Reviewer 2: icpx is not supported on windows as far as I know

Reviewer: The executable for `icpx` is provided as part of the conda package `dpcpp_win-64` on windows.

Author: If I set CXX=icpx on windows it fails while it works on linux. Probably it needs to be investigated further but for now I want to provide the instructions that work
